notebooks/concepts/Mosviz concept notebook.ipynb

```diff
@@ -0,0 +1,542 @@
+      "text/plain": [
+       "<Figure size 864x432 with 4 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "fake_x = np.linspace(6000, 8000, 1024)\n",
+    "fake_y = np.random.randn(len(fake_x)) + 10*np.exp(-0.001*(fake_x-6563)**2) + fake_x/500\n",
+    "fake_2d = fake_y[np.newaxis] + np.random.randn(100, 1024)\n",
+    "fake_r_im = np.sum((np.mgrid[:128, :128]-64)**2,axis=0)**0.5\n",
+    "fake_im = np.random.randn(128, 128) + np.exp(-0.005*fake_r_im**2)*10\n",
+    "\n",
+    "fig = plt.figure(constrained_layout=True, figsize=(12, 6))\n",
+    "gs = fig.add_gridspec(3, 3)\n",
+    "\n",
+    "ax_im = fig.add_subplot(gs[1:, :1])\n",
+    "ax_im.imshow(fake_im)\n",
+    "ax_im.add_artist(plt.Rectangle((62, 4), 5, 120, edgecolor='red', facecolor='none'))\n",
+    "ax_2d = fig.add_subplot(gs[1, 1:])\n",
+    "ax_2d.imshow(fake_2d)\n",
+    "ax_1d = fig.add_subplot(gs[2, 1:])\n",
+    "ax_1d.plot(fake_x, fake_y)\n",
+    "ax_1d.set_xlim(fake_x[0], fake_x[-1])\n",
+    "ax_table = fig.add_subplot(gs[0, :])\n",
+    "ax_table.text(0.5, 0.5, 'TABLE\\nOF SPECTRA\\n(clicking on a table element updates the view below)', ha='center', va='center')\n",
+    "ax_table.set_xticks([])\n",
+    "ax_table.set_yticks([]);"
```

Comment:
That's a good point I did not fully appreciate: having all the sets of axes together in a single viewer means we can implement the tooling for browsing table in the viewer instead of as top-level plugins. This would also mean that the mosviz viewer would be usable in a 'free glue' mode since most functionality would be contained within the viewer.

Reply:
Hmm, a custom viewer concerns me because there are *definitely* use cases where we'd want to show the sub-components of the viewer separate from each other.  

Put another we, way *want* the user to be able to split things up sometimes, just not on "accident".